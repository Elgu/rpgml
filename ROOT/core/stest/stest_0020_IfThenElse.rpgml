
Output i = counter();

exit( i >= 1 );

if( true )
{
  core.IfThenElse    ite();
  ( i == 0 ) -> ite.in_if;
  23         -> ite.in_then;
  "foo"      -> ite.in_else;

  print( ite.out + "\n" );
}

if( true )
{
  print( core.toString( i == 0 ? 23 : "foo" ) + "\n" );
}

if( true )
{
  Output c = [ 0, 1, 2 ] * (counter()+1);

  Output t = [ 1.5, 3.5, 1.25 ];
  Output e = [ 2  , 3  , 4    ];

  core.IfThenElse    ite();
  ( c >= 2 ) -> ite.in_if;
  t -> ite.in_then;
  e -> ite.in_else;

  print( ite.out );
  print( "\n" );
}

if( true )
{
  Output c = [ 0, 1, 2 ] * (counter()+1);

  Output t = [ 1.5, 3.5, 1.25 ];
  Output e = [ 2  , 3  , 4    ];

  print( core.toString( c >= 2 ? t : e ) + "\n" );
}

if( true )
{
  Output c = [ 1, 2, 3, 4 ] + counter();

  Output t = [ "foo", "bar", "bat", "baz" ];
  Output e = "false";

  core.IfThenElse    ite();
  bool( c % 2 ) -> ite.in_if;
  t -> ite.in_then;
  e -> ite.in_else;

  print( ite.out );
  print( "\n" );
}

if( true )
{
  Output c = [ 1, 2, 3, 4 ] + counter();

  Output t = [ "foo", "bar", "bat", "baz" ];
  Output e = "false";

  print( core.toString( c % 2 ? t : e ) + "\n" );
}

if( true )
{
  Output c = [ 0, 1, 2 ] + counter();

  Output t = 11.5;
  Output e = [ 2  , 3  , 4    ];

  core.IfThenElse    ite();
  bool( c % 2 ) -> ite.in_if;
  t -> ite.in_then;
  e -> ite.in_else;

  print( ite.out );
  print( "\n" );
}

if( true )
{
  Output c = [ 0, 1, 2 ] + counter();

  Output t = 11.5;
  Output e = [ 2  , 3  , 4    ];

  print( core.toString( c % 2 ? t : e ) + "\n" );
}

if( true )
{
  Output c = [ 1, 2, 3 ] + counter();

  Output t = [ 1.5, 3.5, 1.25 ];
  Output e = 42;

  core.IfThenElse    ite();
  bool( c % 2 ) -> ite.in_if;
  t -> ite.in_then;
  e -> ite.in_else;

  print( ite.out );
  print( "\n" );
}

if( true )
{
  Output c = [ 1, 2, 3 ] + counter();

  Output t = [ 1.5, 3.5, 1.25 ];
  Output e = 42;

  print( core.toString( c % 2 ? t : e ) + "\n" );
}

