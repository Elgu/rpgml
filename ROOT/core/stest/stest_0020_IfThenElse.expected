23
23
[ 2, 3, 1.25 ]
[ 2, 3, 1.25 ]
[ "foo", "false", "bat", "false" ]
[ "foo", "false", "bat", "false" ]
[ 2, 11.5, 4 ]
[ 2, 11.5, 4 ]
[ 1.5, 42, 1.25 ]
[ 1.5, 42, 1.25 ]
foo
foo
[ 2, 3.5, 1.25 ]
[ 2, 3.5, 1.25 ]
[ "false", "bar", "false", "baz" ]
[ "false", "bar", "false", "baz" ]
[ 11.5, 3, 11.5 ]
[ 11.5, 3, 11.5 ]
[ 42, 3.5, 42 ]
[ 42, 3.5, 42 ]
